include/etl/uncopyable.h

```diff
@@ -0,0 +1,56 @@
+///\file
+
+/******************************************************************************
+The MIT License(MIT)
+
+Embedded Template Library.
+https://github.com/ETLCPP/etl
+https://www.etlcpp.com
+
+Copyright(c) 2024 BMW AG
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files(the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions :
+
+The above copyright notice and this permission notice shall be included in all
+copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+******************************************************************************/
+
+#ifndef ETL_UNCOPYABLE_INCLUDED
+#define ETL_UNCOPYABLE_INCLUDED
+
+#include "platform.h"
+
+namespace etl
+{
+#if ETL_USING_CPP11
+  //***************************************************************************
+  /// An uncopyable base class.
+  /// Can be used to make a class uncopyable by deriving from it.
+  //***************************************************************************
+  class uncopyable
+  {
+  public:
+    uncopyable(uncopyable const&)            = delete;
+    uncopyable& operator=(uncopyable const&) = delete;
+
+  protected:
+    uncopyable() = default;
```

Comment:
I've see comments online that Scott Meyers also adds the destructor in his 'Effective C++' book.

Reply:
I.e. a non-virtual public "=default" destructor?

Is there a reasoning behind it?

Reply:
It looks like it is intended to make the destructor `protected`, rather than default `public`.

Reply:
Added the destructor.